docs/user/automatic-redirects.rst

```diff
@@ -1,56 +1,140 @@
+You might be familiar with permalinks from blogging.
+The idea is that a blog post receives a unique link as soon as it's published and that the link does not change afterward.
+Incoming sources can reference the blog post even though the blog changes structure or the post title changes.
 
-Among all the multiple versions of docs, you can choose which is the "default" version for RTD to display, which usually corresponds to the git branch of the most recent official release from your project.
+When creating an external link to a specific documentation page,
+chances are that the page is moved as the documentation changes over time.
 
-rtfd.io
-~~~~~~~~
+How should a permalink look for a documentation project?
+Firstly, you should know that a *permalink* does not really exist in documentation but it is the result of careful actions to avoid breaking incoming links.
 
-Links to rtfd.io are treated the same way as above.
+As a documentation owner,
+you most likely want users clicking on incoming links to see the latest version of the page.
+
+Good practice ✅
+~~~~~~~~~~~~~~~~
+
+* Use ``/page/path/to/page.html`` if you are linking to the page in the default version of the default language.
+* Use ``/en/latest/path/to/page.html`` if you want to be specific about the language.
+* If you move a page that likely has incoming references, :doc:`create a redirect rule </user-defined-redirects>`.
+* Links from other Sphinx projects should use :doc:`intersphinx </guides/intersphinx>`.
+* Use minimal page slugs that do not attract renaming suggestions.
+* When possible,
+  keep original file names rather than going for low-impact URL renaming.
+  Renaming an article's title is great for the reader and great for SEO,
+  but this does not have to involve the URL.
+
+Questionable practice 🟡
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Avoid using specific versions in links unless you really mean that users should see that exact version.
+  Versions get outdated.
+  You can use a :ref:`version warning <versions:Version warning>` to ensure the reader is aware.
+* Keep development versions hidden so people do not find them on search engines by mistake.
+* Avoid using a public ``latest`` for development versions and do not make your *default version* a development branch.
+  Publishing development branches can mean that users are reading instructions for unreleased software or draft documentation.
+* Try to get your understanding of ``latest`` and *default version* right from the beginning and ensure you don't change them later on.
```

Comment:
This also seems to be good advice. I think I'm a little confused about this "Questionable practice" section 🤔

Reply:
I'll put the end of the section at the beginning :+1:

Reply:
I moved it to best practice :)